bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -710,6 +714,24 @@ impl Timeline {
     }
 }
 
+/// Given a pair of optional `caption` and `formatted_caption` parameters,
+/// return a formatted caption:
+///
+/// - If a `formatted_caption` exists, return it.
+/// - If it doesn't exist but there is a `caption`, parse it as markdown and
+///   return the result.
+/// - Return `None` if there are no `caption` or `formatted_caption` parameters.
+fn formatted_caption_from(
+    caption: &Option<String>,
+    formatted_caption: &Option<FormattedBody>,
```

Comment:
nit: don't pass `formatted_caption` by ref, since it must be consumed by this function; avoids a `clone()` below